src/main/scala/com/fulcrumgenomics/fasta/SequenceInfos.scala

```diff
@@ -0,0 +1,171 @@
+  require(length > 0, s"Length must be > 0 for '$name'")
+
+  /** All names, including aliases */
+  @inline final def allNames: Seq[String] = name +: aliases
+  @inline final def apply[T](key: String): T = this.attributes(key).asInstanceOf[T]
+  @inline final def get[T](key: String): Option[T] = this.attributes.get(key).map(_.asInstanceOf[T])
+  @inline final def contains(key: String): Boolean = this.attributes.contains(key)
+
+  @inline final def isAlternate: Boolean = this.contains("AH")
+  def alternate: Option[AlternateLocus] = {
+    this.apply[String]("AH") match {
+      case "*" => None
+      case string =>
+        val Array(refName, rest) = string.split(':')
+        val Array(start, end) = rest.split('-')
+        val locus = refName match {
+          case "=" => AlternateLocus(refName=this.name, start=start.toInt, end=end.toInt)
+          case _   => AlternateLocus(refName=refName, start=start.toInt, end=end.toInt)
+        }
+        Some(locus)
+    }
+  }
+
+  @inline final def m5: String = this.apply(SAMSequenceRecord.MD5_TAG)
+  @inline final def m5Int: BigInt = BigInt(this.m5, 16)
+
+  @inline final def assembly: String = this.apply(SAMSequenceRecord.ASSEMBLY_TAG)
+  @inline final def uri: String = this.apply(SAMSequenceRecord.URI_TAG)
+  @inline final def species: String = this.apply(SAMSequenceRecord.SPECIES_TAG)
+  @inline final def description: String = this.apply(SAMSequenceRecord.DESCRIPTION_TAG)
+  @inline final def topology: Topology = Topology.values.find(_.name == this.apply("TP")).get
+
+  /** Returns true if the the sequences share a common reference name (including aliases), have the same length, and
+    * the same MD5 if both have MD5s. */
+  def same(that: SequenceInfo): Boolean = {
+    val thisM5: Option[BigInt] = if (this.contains(SAMSequenceRecord.MD5_TAG)) Some(this.m5Int) else None
+    val thatM5: Option[BigInt] = if (that.contains(SAMSequenceRecord.MD5_TAG)) Some(that.m5Int) else None
+
+    this.allNames.exists(that.allNames.contains) &&
+      this.length == that.length &&
+      thisM5 == thatM5
+  }
+
+}
+
+/** Contains an ordered collection of sequences. */
+class SequenceInfos(infos: IndexedSeq[SequenceInfo]) extends Iterable[SequenceInfo] {
+  // Ensure that all names, even aliases, are unique.
+  {
+    val duplicateNames: String = infos.flatMap(_.allNames)
+      .groupBy(identity)
+      .filter(_._2.length > 1)
+      .keys
+      .mkString(", ")
+    require(duplicateNames.isEmpty, f"Found duplicate names: $duplicateNames")
+  }
+
+  private val mapping: Map[String, SequenceInfo] = infos.flatMap { info =>
+    info.allNames.map { name => name -> info }
+  }.toMap
+
+  def apply(name: String): SequenceInfo = this.mapping(name)
+  def get(name: String): Option[SequenceInfo] = this.mapping.get(name)
+  def contains(name: String): Boolean = this.mapping.contains(name)
+  override def iterator: Iterator[SequenceInfo] = this.infos.iterator
+}
```

Comment:
I know `infos` is public, but I think we'd still want a `def apply(index: int) = this.info(int)`